cpp/arcticdb/version/version_map_batch_methods.hpp

```diff
@@ -192,8 +192,14 @@ inline std::optional<AtomKey> get_key_for_version_query(
         [&version_map_entry] (const pipelines::SpecificVersionQuery& specific_version) {
             return find_index_key_for_version_id(specific_version.version_id_, version_map_entry);
         },
-        [&version_map_entry] (const pipelines::TimestampVersionQuery& timestamp_version) {
-        return find_index_key_for_version_timestamp(timestamp_version.timestamp_, version_map_entry);
+        [&version_map_entry] (const pipelines::TimestampVersionQuery& timestamp_version) -> std::optional<AtomKey> {
+            auto version_key = get_version_key_from_time_for_versions(timestamp_version.timestamp_, version_map_entry->get_indexes(false));
+            if(version_key.has_value()){
+                auto version_id = version_key.value().version_id();
```

Comment:
Could you please type `version_key` and `version_id` explicitly? :pray:

Reply:
we prefer auto  as a style in the code base. I will politely dismiss your comment here